runtime/src/main/jni/Runtime.cpp

```diff
@@ -301,9 +309,39 @@ jint Runtime::GenerateNewObjectId(JNIEnv *env, jobject obj)
 	return objectId;
 }
 
-void Runtime::AdjustAmountOfExternalAllocatedMemoryNative(JNIEnv *env, jobject obj, jlong usedMemory)
+void Runtime::AdjustAmountOfExternalAllocatedMemory() {
+	int64_t usedMemory = m_env.CallLongMethod(m_runtime, GET_USED_MEMORY_METHOD_ID);
+	int64_t changeInBytes = usedMemory - m_lastUsedMemory;
+	int64_t externalMemory = 0;
+
+	if (changeInBytes != 0) {
+		externalMemory = m_isolate->AdjustAmountOfExternalAllocatedMemory(changeInBytes);
+	}
+
+	DEBUG_WRITE("usedMemory=%" PRId64 " changeInBytes=%" PRId64 " externalMemory=%" PRId64, usedMemory, changeInBytes, externalMemory);
+
+	m_lastUsedMemory = usedMemory;
+}
+
+bool Runtime::NotifyGC(JNIEnv *env, jobject obj)
 {
-	m_isolate->AdjustAmountOfExternalAllocatedMemory(usedMemory);
+	bool success = __sync_bool_compare_and_swap(&m_runGC, false, true);
+	return success;
+}
+
+bool Runtime::TryCallGC() {
```

Comment:
seems like its depreciated with IdleNotificationDeadline and LowMemoryNotification methods. How about using these instead of interveening with v8 GC